src/components/Asset/AssetActions/Trade/Output.tsx

```diff
@@ -31,12 +34,14 @@ export default function Output({
     if (!poolAddress || !isAssetNetwork) return
 
     async function getSwapFee() {
-      // const swapFee = await ocean.pool.getSwapFee(poolAddress)
+      if (!web3) return
+      const poolInstance = new Pool(web3)
+      const swapFee = await poolInstance.getSwapFee(poolAddress)
```

Comment:
mmmh, we should not rely on web3 calls to get this but use the subgraph instead, as this should always work, no matter if wallet is connected or not.

But if I remember correctly, fees are not fully implemented yet in subgraph so will always return 0 when queried so for time being, web3 calls are only way to make it work at least

Reply:
you don't even need this.